docs/usage.rst

```diff
@@ -0,0 +1,98 @@
+*****
+Usage
+*****
+
+Run Hyperparameter Search
+=========================
+
+cluster_utils provides two main commands to run hyperparameter search in different
+modes:
+
+- ``grid_search``:  Simple grid search over specified hyperparameter ranges.
+- ``hp_optimization``:  Uses sampling-based optimization to search for best combination
+  of hyperparameters within the specified ranges.
+
+They are both run as modules via ``python -m`` and expect a configuration file as
+argument:
+
+.. code-block:: bash
+
+   python3 -m cluster_utils.grid_search config_for_grid_search.json
+
+   python3 -m cluster_utils.hp_optimization config_for_hp_optim.json
+
+See :doc:`configuration` for information on the expected structure of the config file
+(note that there are differences between the two methods!).
+
+Optionally, a list of key-value arguments can be provided in addition, to overwrite
+single settings from the config file.  Use dot-notation to specify nested keys.  Example:
+
+.. code-block:: bash
+
+   python3 -m cluster_utils.grid_search config.json \
+     'results_dir="/tmp"' 'optimization_setting.run_local=True'
```

Comment:
Is `optimization_setting.run_local` a setting we actually support (if it is, I missed that)? We should only list commands that directly work when copied.

Reply:
Yes, that is an actual setting.  It suppresses the "Do you want to run locally?" prompt when running locally (see [docs](https://martius-lab.github.io/cluster_utils/configuration.html#confval-optimization_setting.run_local).
However, I now realised that this is for some reason an hp_optimization-specific setting, i.e. it doesn't work with grid_search.  I'll change the example accordingly.